error running file

Home Assistant Core 2024.10.1
ESPHome Current version: 2024.9.2

I configure esp32 c3 super mini as follows
## this code yaml
```
esphome:
  name: ld2412-esp32-c3-new
  comment: LD2412 ESP32 C3 NEW
  friendly_name: LD2412 ESP32 C3 NEW
  name_add_mac_suffix: True
  platformio_options:
    board_build.flash_mode: dio
  project: 
    name: Dev.Human_Presence_Sensor
    version: LD2412
logger:
preferences:
    flash_write_interval: 5s
esp32:
  board: lolin_c3_mini
  framework:
    type: arduino
    version: 2.0.9
    platform_version: 6.3.0
uart:
  id: uart_bus
  tx_pin: 
    number: GPIO3
    mode:
      input: true
      pullup: true
  rx_pin: 
    number: GPIO2
    mode:
      input: true
      pullup: true
  baud_rate: 115200
  parity: NONE
  stop_bits: 1
  data_bits: 8
improv_serial:
debug:
  update_interval: 30s
api:
ota:
  - platform: esphome
wifi:
  power_save_mode: LIGHT
  reboot_timeout: 10min
  ap:
    ssid: "HUMAN SENSOR LD2412"
captive_portal:
web_server:
  port: 80
external_components:
  - source:
      type: git
      url: https://github.com/Rihan9/LD2412
      ref: main
    components: [LD2412]
LD2412:
  id: ld2412
  throttle: 3s

binary_sensor:
  - platform: LD2412
    has_target:
      name: Presence
    has_moving_target:
      name: Moving Target
    has_still_target:
      name: Still Target
sensor:
  - platform: LD2412
    moving_distance:
      name : Moving Distance
    still_distance:
      name: Still Distance
    moving_energy:
      name: Move Energy
    still_energy:
      name: Still Energy
    detection_distance:
      name: Detection Distance
    light:
      name: light
    g0:
      move_energy:
        name: g00 move energy
      still_energy:
        name: g00 still energy
    g1:
      move_energy:
        name: g01 move energy
      still_energy:
        name: g01 still energy
    g2:
      move_energy:
        name: g02 move energy
      still_energy:
        name: g02 still energy
    g3:
      move_energy:
        name: g03 move energy
      still_energy:
        name: g03 still energy
    g4:
      move_energy:
        name: g04 move energy
      still_energy:
        name: g04 still energy
    g5:
      move_energy:
        name: g05 move energy
      still_energy:
        name: g05 still energy
    g6:
      move_energy:
        name: g06 move energy
      still_energy:
        name: g06 still energy
    g7:
      move_energy:
        name: g07 move energy
      still_energy:
        name: g07 still energy
    g8:
      move_energy:
        name: g08 move energy
      still_energy:
        name: g08 still energy
    g9:
      move_energy:
        name: g09 move energy
      still_energy:
        name: g09 still energy
    g10:
      move_energy:
        name: g10 move energy
      still_energy:
        name: g10 still energy
    g11:
      move_energy:
        name: g11 move energy
      still_energy:
        name: g11 still energy
    g12:
      move_energy:
        name: g12 move energy
      still_energy:
        name: g12 still energy
    g13:
      move_energy:
        name: g13 move energy
      still_energy:
        name: g13 still energy
number:
  - platform: LD2412
    timeout:
      name: "presence holding"
    min_distance_gate:
      name: "min distance gate"
    max_distance_gate:
      name: "max distance gate"
    g0:
      move_threshold:
        name: g00 move threshold
      still_threshold:
        name: g00 still threshold
    g1:
      move_threshold:
        name: g01 move threshold
      still_threshold:
        name: g01 still threshold
    g2:
      move_threshold:
        name: g02 move threshold
      still_threshold:
        name: g02 still threshold
    g3:
      move_threshold:
        name: g03 move threshold
      still_threshold:
        name: g03 still threshold
    g4:
      move_threshold:
        name: g04 move threshold
      still_threshold:
        name: g04 still threshold
    g5:
      move_threshold:
        name: g05 move threshold
      still_threshold:
        name: g05 still threshold
    g6:
      move_threshold:
        name: g06 move threshold
      still_threshold:
        name: g06 still threshold
    g7:
      move_threshold:
        name: g07 move threshold
      still_threshold:
        name: g07 still threshold
    g8:
      move_threshold:
        name: g08 move threshold
      still_threshold:
        name: g08 still threshold
    g9:
      move_threshold:
        name: g09 move threshold
      still_threshold:
        name: g09 still threshold
    g10:
      move_threshold:
        name: g10 move threshold
      still_threshold:
        name: g10 still threshold
    g11:
      move_threshold:
        name: g11 move threshold
      still_threshold:
        name: g11 still threshold
    g12:
      move_threshold:
        name: g12 move threshold
      still_threshold:
        name: g12 still threshold
    g13:
      move_threshold:
        name: g13 move threshold
      still_threshold:
        name: g13 still threshold
select:
  - platform: LD2412
    out_pin_level:
      name: 'Hardware output pin level'
    distance_resolution:
      name: 'Distance resolution'
    baud_rate:
      name: "baud rate"
      on_value:
        - delay: 3s
        - lambda: |-
            id(uart_bus).flush();
            uint32_t new_baud_rate = stoi(x);
            ESP_LOGD("change_baud_rate", "Changing baud rate from %i to %i",id(uart_bus).get_baud_rate(), new_baud_rate);
            if (id(uart_bus).get_baud_rate() != new_baud_rate) {
            id(uart_bus).set_baud_rate(new_baud_rate);
            id(uart_bus).load_settings();
            }
    mode:
      name: "Mode"
button:
  - platform: LD2412
    factory_reset:
      name: "factory reset"
    restart:
      name: "restart"
    query_params:
      name: query params

text_sensor:
  - platform: LD2412
    version:
      name: "firmware version"
    mac_address:
      name: "mac address"
switch:
  - platform: LD2412
    bluetooth:
      name: "Bluetooth"
```
## and when running it has the following error
```
INFO ESPHome 2024.9.2
INFO Reading configuration /config/esphome/ld2412-esp32-c3-new.yaml...
WARNING The selected Arduino framework version is not the recommended one. If there are connectivity or build issues please remove the manual version.
WARNING The selected Arduino framework version is not the recommended one. If there are connectivity or build issues please remove the manual version.
WARNING GPIO2 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
INFO Generating C++ source...
INFO Compiling app...
Processing ld2412-esp32-c3-new (board: lolin_c3_mini; framework: arduino; platform: platformio/espressif32@6.3.0)

HARDWARE: ESP32C3 160MHz, 320KB RAM, 4MB Flash
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5
Dependency Graph
|-- AsyncTCP-esphome @ 2.1.4
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 3.2.2
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- ArduinoJson @ 6.18.5
|-- Improv @ 1.2.4
Compiling .pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/LD2412.cpp.o
Compiling .pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/query_button.cpp.o
Compiling .pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/reset_button.cpp.o
Compiling .pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/restart_button.cpp.o
In file included from /data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Stream.h:26,
                 from /data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:177,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson/Configuration.hpp:156,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.hpp:7,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.h:9,
                 from src/esphome/components/json/json_util.h:11,
                 from src/esphome/core/string_ref.h:10,
                 from src/esphome/core/entity_base.h:5,
                 from src/esphome/components/button/button.h:4,
                 from src/esphome/components/LD2412/button/query_button.h:3,
                 from src/esphome/components/LD2412/button/query_button.cpp:1:
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Print.h:30:13: error: expected unqualified-id before numeric constant
 #define HEX 16
             ^~
src/esphome/components/LD2412/button/../LD2412.h:127:19: note: in expansion of macro 'HEX'
 static const char HEX[] = "0123456789ABCDEF";
                   ^~~
In file included from src/esphome/components/LD2412/button/query_button.h:4,
                 from src/esphome/components/LD2412/button/query_button.cpp:1:
src/esphome/components/LD2412/button/../LD2412.h: In member function 'std::__cxx11::string esphome::LD2412::LD2412Component::format_buffer(uint8_t*, size_t)':
src/esphome/components/LD2412/button/../LD2412.h:248:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 0] = HEX[((buffer[i] & 0xF0) >> 4)];
                                                           ^
src/esphome/components/LD2412/button/../LD2412.h:249:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 1] = HEX[((buffer[i] & 0x0F) >> 0)];
                                                           ^
In file included from /data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Stream.h:26,
                 from /data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:177,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson/Configuration.hpp:156,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.hpp:7,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.h:9,
                 from src/esphome/components/json/json_util.h:11,
                 from src/esphome/core/string_ref.h:10,
                 from src/esphome/core/entity_base.h:5,
                 from src/esphome/components/button/button.h:4,
                 from src/esphome/components/LD2412/button/restart_button.h:3,
                 from src/esphome/components/LD2412/button/restart_button.cpp:1:
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Print.h:30:13: error: expected unqualified-id before numeric constant
 #define HEX 16
             ^~
src/esphome/components/LD2412/button/../LD2412.h:127:19: note: in expansion of macro 'HEX'
 static const char HEX[] = "0123456789ABCDEF";
                   ^~~
In file included from /data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Stream.h:26,
                 from /data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:177,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson/Configuration.hpp:156,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.hpp:7,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.h:9,
                 from src/esphome/components/json/json_util.h:11,
                 from src/esphome/core/string_ref.h:10,
                 from src/esphome/core/entity_base.h:5,
                 from src/esphome/components/button/button.h:4,
                 from src/esphome/components/LD2412/button/reset_button.h:3,
                 from src/esphome/components/LD2412/button/reset_button.cpp:1:
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Print.h:30:13: error: expected unqualified-id before numeric constant
 #define HEX 16
             ^~
src/esphome/components/LD2412/button/../LD2412.h:127:19: note: in expansion of macro 'HEX'
 static const char HEX[] = "0123456789ABCDEF";
                   ^~~
In file included from src/esphome/components/LD2412/button/reset_button.h:4,
                 from src/esphome/components/LD2412/button/reset_button.cpp:1:
src/esphome/components/LD2412/button/../LD2412.h: In member function 'std::__cxx11::string esphome::LD2412::LD2412Component::format_buffer(uint8_t*, size_t)':
src/esphome/components/LD2412/button/../LD2412.h:248:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 0] = HEX[((buffer[i] & 0xF0) >> 4)];
                                                           ^
src/esphome/components/LD2412/button/../LD2412.h:249:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 1] = HEX[((buffer[i] & 0x0F) >> 0)];
                                                           ^
In file included from src/esphome/components/LD2412/button/restart_button.h:4,
                 from src/esphome/components/LD2412/button/restart_button.cpp:1:
src/esphome/components/LD2412/button/../LD2412.h: In member function 'std::__cxx11::string esphome::LD2412::LD2412Component::format_buffer(uint8_t*, size_t)':
src/esphome/components/LD2412/button/../LD2412.h:248:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 0] = HEX[((buffer[i] & 0xF0) >> 4)];
                                                           ^
src/esphome/components/LD2412/button/../LD2412.h:249:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 1] = HEX[((buffer[i] & 0x0F) >> 0)];
                                                           ^
In file included from /data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Stream.h:26,
                 from /data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:177,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson/Configuration.hpp:156,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.hpp:7,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.h:9,
                 from src/esphome/components/json/json_util.h:11,
                 from src/esphome/core/string_ref.h:10,
                 from src/esphome/core/entity_base.h:5,
                 from src/esphome/components/binary_sensor/binary_sensor.h:4,
                 from src/esphome/components/LD2412/LD2412.h:5,
                 from src/esphome/components/LD2412/LD2412.cpp:1:
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/Print.h:30:13: error: expected unqualified-id before numeric constant
 #define HEX 16
             ^~
src/esphome/components/LD2412/LD2412.h:127:19: note: in expansion of macro 'HEX'
 static const char HEX[] = "0123456789ABCDEF";
                   ^~~
In file included from src/esphome/components/LD2412/LD2412.cpp:1:
src/esphome/components/LD2412/LD2412.h: In member function 'std::__cxx11::string esphome::LD2412::LD2412Component::format_buffer(uint8_t*, size_t)':
src/esphome/components/LD2412/LD2412.h:248:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 0] = HEX[((buffer[i] & 0xF0) >> 4)];
                                                           ^
src/esphome/components/LD2412/LD2412.h:249:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 1] = HEX[((buffer[i] & 0x0F) >> 0)];
                                                           ^
*** [.pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/query_button.cpp.o] Error 1
*** [.pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/reset_button.cpp.o] Error 1
*** [.pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/restart_button.cpp.o] Error 1
*** [.pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/LD2412.cpp.o] Error 1
========================= [FAILED] Took 14.18 seconds =========================
```

thank You. But after I removed those 2 parameters, the following error appeared.

```
INFO ESPHome 2024.9.2
INFO Reading configuration /config/esphome/ld2412-esp32-c3-new.yaml...
WARNING GPIO2 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
INFO Generating C++ source...
INFO Compiling app...
Processing ld2412-esp32-c3-new (board: lolin_c3_mini; framework: arduino; platform: platformio/espressif32@5.4.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32C3 160MHz, 320KB RAM, 4MB Flash
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5
Dependency Graph
|-- AsyncTCP-esphome @ 2.1.4
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 3.2.2
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- ArduinoJson @ 6.18.5
|-- Improv @ 1.2.4
Compiling .pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/LD2412.cpp.o
Compiling .pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/query_button.cpp.o
Compiling .pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/reset_button.cpp.o
Compiling .pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/restart_button.cpp.o
In file included from /data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Stream.h:26,
                 from /data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Arduino.h:167,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson/Configuration.hpp:156,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.hpp:7,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.h:9,
                 from src/esphome/components/json/json_util.h:11,
                 from src/esphome/core/string_ref.h:10,
                 from src/esphome/core/entity_base.h:5,
                 from src/esphome/components/button/button.h:4,
                 from src/esphome/components/LD2412/button/reset_button.h:3,
                 from src/esphome/components/LD2412/button/reset_button.cpp:1:
/data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Print.h:30:13: error: expected unqualified-id before numeric constant
 #define HEX 16
             ^~
src/esphome/components/LD2412/button/../LD2412.h:127:19: note: in expansion of macro 'HEX'
 static const char HEX[] = "0123456789ABCDEF";
                   ^~~
In file included from /data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Stream.h:26,
                 from /data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Arduino.h:167,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson/Configuration.hpp:156,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.hpp:7,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.h:9,
                 from src/esphome/components/json/json_util.h:11,
                 from src/esphome/core/string_ref.h:10,
                 from src/esphome/core/entity_base.h:5,
                 from src/esphome/components/button/button.h:4,
                 from src/esphome/components/LD2412/button/query_button.h:3,
                 from src/esphome/components/LD2412/button/query_button.cpp:1:
/data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Print.h:30:13: error: expected unqualified-id before numeric constant
 #define HEX 16
             ^~
src/esphome/components/LD2412/button/../LD2412.h:127:19: note: in expansion of macro 'HEX'
 static const char HEX[] = "0123456789ABCDEF";
                   ^~~
In file included from src/esphome/components/LD2412/button/query_button.h:4,
                 from src/esphome/components/LD2412/button/query_button.cpp:1:
src/esphome/components/LD2412/button/../LD2412.h: In member function 'std::__cxx11::string esphome::LD2412::LD2412Component::format_buffer(uint8_t*, size_t)':
src/esphome/components/LD2412/button/../LD2412.h:248:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 0] = HEX[((buffer[i] & 0xF0) >> 4)];
                                                           ^
In file included from src/esphome/components/LD2412/button/reset_button.h:4,
                 from src/esphome/components/LD2412/button/reset_button.cpp:1:
src/esphome/components/LD2412/button/../LD2412.h: In member function 'std::__cxx11::string esphome::LD2412::LD2412Component::format_buffer(uint8_t*, size_t)':
src/esphome/components/LD2412/button/../LD2412.h:248:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 0] = HEX[((buffer[i] & 0xF0) >> 4)];
                                                           ^
src/esphome/components/LD2412/button/../LD2412.h:249:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 1] = HEX[((buffer[i] & 0x0F) >> 0)];
                                                           ^
src/esphome/components/LD2412/button/../LD2412.h:249:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 1] = HEX[((buffer[i] & 0x0F) >> 0)];
                                                           ^
In file included from /data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Stream.h:26,
                 from /data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Arduino.h:167,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson/Configuration.hpp:156,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.hpp:7,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.h:9,
                 from src/esphome/components/json/json_util.h:11,
                 from src/esphome/core/string_ref.h:10,
                 from src/esphome/core/entity_base.h:5,
                 from src/esphome/components/button/button.h:4,
                 from src/esphome/components/LD2412/button/restart_button.h:3,
                 from src/esphome/components/LD2412/button/restart_button.cpp:1:
/data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Print.h:30:13: error: expected unqualified-id before numeric constant
 #define HEX 16
             ^~
src/esphome/components/LD2412/button/../LD2412.h:127:19: note: in expansion of macro 'HEX'
 static const char HEX[] = "0123456789ABCDEF";
                   ^~~
In file included from src/esphome/components/LD2412/button/restart_button.h:4,
                 from src/esphome/components/LD2412/button/restart_button.cpp:1:
src/esphome/components/LD2412/button/../LD2412.h: In member function 'std::__cxx11::string esphome::LD2412::LD2412Component::format_buffer(uint8_t*, size_t)':
src/esphome/components/LD2412/button/../LD2412.h:248:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 0] = HEX[((buffer[i] & 0xF0) >> 4)];
                                                           ^
src/esphome/components/LD2412/button/../LD2412.h:249:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 1] = HEX[((buffer[i] & 0x0F) >> 0)];
                                                           ^
*** [.pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/reset_button.cpp.o] Error 1
*** [.pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/query_button.cpp.o] Error 1
In file included from /data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Stream.h:26,
                 from /data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Arduino.h:167,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson/Configuration.hpp:156,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.hpp:7,
                 from .piolibdeps/ld2412-esp32-c3-new/ArduinoJson/src/ArduinoJson.h:9,
                 from src/esphome/components/json/json_util.h:11,
                 from src/esphome/core/string_ref.h:10,
                 from src/esphome/core/entity_base.h:5,
                 from src/esphome/components/binary_sensor/binary_sensor.h:4,
                 from src/esphome/components/LD2412/LD2412.h:5,
                 from src/esphome/components/LD2412/LD2412.cpp:1:
/data/cache/platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/Print.h:30:13: error: expected unqualified-id before numeric constant
 #define HEX 16
             ^~
src/esphome/components/LD2412/LD2412.h:127:19: note: in expansion of macro 'HEX'
 static const char HEX[] = "0123456789ABCDEF";
                   ^~~
In file included from src/esphome/components/LD2412/LD2412.cpp:1:
src/esphome/components/LD2412/LD2412.h: In member function 'std::__cxx11::string esphome::LD2412::LD2412Component::format_buffer(uint8_t*, size_t)':
src/esphome/components/LD2412/LD2412.h:248:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 0] = HEX[((buffer[i] & 0xF0) >> 4)];
                                                           ^
src/esphome/components/LD2412/LD2412.h:249:59: error: invalid types 'int[int]' for array subscript
       version[(i * 2) + 1] = HEX[((buffer[i] & 0x0F) >> 0)];
                                                           ^
*** [.pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/button/restart_button.cpp.o] Error 1
*** [.pioenvs/ld2412-esp32-c3-new/src/esphome/components/LD2412/LD2412.cpp.o] Error 1
========================= [FAILED] Took 14.68 seconds =========================
```

--- a/components/LD2412/LD2412.h
+++ b/components/LD2412/LD2412.h
@@ -124,7 +124,7 @@ static const uint8_t CMD_FRAME_END[4] = {0x04, 0x03, 0x02, 0x01};
 static const uint8_t DATA_FRAME_HEADER[4] = {0xF4, 0xF3, 0xF2, 0xF1};
 static const uint8_t DATA_FRAME_END[4] = {0xF8, 0xF7, 0xF6, 0xF5};
 
-static const char HEX[] = "0123456789ABCDEF"; 
+static const char HEX_POSITIONING_CONVERSION[] = "0123456789ABCDEF"; 
 
 /*
 Data Type: 6th byte
@@ -245,8 +245,8 @@ class LD2412Component : public Component, public uart::UARTDevice {
     version.resize(len*2+1);
     for (size_t i = 0; i < len; i++)
     {
-      version[(i * 2) + 0] = HEX[((buffer[i] & 0xF0) >> 4)];
-      version[(i * 2) + 1] = HEX[((buffer[i] & 0x0F) >> 0)];
+      version[(i * 2) + 0] = HEX_POSITIONING_CONVERSION[((buffer[i] & 0xF0) >> 4)];
+      version[(i * 2) + 1] = HEX_POSITIONING_CONVERSION[((buffer[i] & 0x0F) >> 0)];
     }
     return version;
   }

